packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -85,123 +109,506 @@ const propTypes = {
+    if (DateUtil.isValidDate(value, momentDateFormat)) {
+      const parseDate = DateUtil.getDateInputValues(DateUtil.dateOrder.YMD, dateValue, '-');
+      dateDispatch(parseDate);
+    } else if (dateValue === '') {
+      dateDispatch({ day: '', month: '', year: '' });
+    }
+  }, [momentDateFormat, dateValue, value]);
+
+  /**
+   * Moves focus to the correct input depending on date ordering. Focus changing is
+   * disabled if a complete date has been entered in order to make single input
+   * corrections easier, and is re-enabled if the whole date is erased.
+   * @param {string} inputValue - The value from the current input.
+   * @param {number} type - The input type, based on DateUtil.inputType.
+   * @param {string} dateOrder - Date variant based on the currently supported date orders: 'MDY', 'DMY', or 'YMD'.
+   */
+  const moveFocusOnChange = (inputValue, type, dateOrder) => {
+    if (dateOrder === DateUtil.dateOrder.MDY) {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.MONTH) {
+          dayInputRef.current.focus();
+        } else {
+          yearInputRef.current.focus();
+        }
+      }
+    } else if (dateOrder === DateUtil.dateOrder.DMY) {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.DAY) {
+          monthInputRef.current.focus();
+        } else {
+          yearInputRef.current.focus();
+        }
+      }
+    } else if (dateOrder === DateUtil.dateOrder.YMD) {
+      if (inputValue.length === 4) {
+        monthInputRef.current.focus();
+      } else if (inputValue.length === 2 && type === DateUtil.inputType.MONTH) {
+        dayInputRef.current.focus();
+      }
+    }
+  };
+
+  /**
+   * Sets state for the day, month, and year.
+   * @param {string} inputValue - The input value to set in state.
+   * @param {number} type - The inputType (day, month, or year).
+   */
+  const setDate = (inputValue, type) => {
+    if (type === DateUtil.inputType.DAY) {
+      dateDispatch({ day: inputValue });
+    } else if (type === DateUtil.inputType.MONTH) {
+      dateDispatch({ month: inputValue });
+    } else {
+      dateDispatch({ year: inputValue });
+    }
+
+    moveFocusOnChange(inputValue, type, dateFormatOrder);
+  };
 
-    this.uuid = uuidv4();
+  const handleDateChange = (event, inputValue, type) => {
+    /**
+     * Sets the day, month and year based on input values, formats them
+     * based on the date format variant, and passes the formatted date to onChange.
+     */
+    const day = type === DateUtil.inputType.DAY ? inputValue : date.day;
+    const month = type === DateUtil.inputType.MONTH ? inputValue : date.month;
+    const year = type === DateUtil.inputType.YEAR ? inputValue : date.year;
+
+    if (type === DateUtil.inputType.DAY) {
+      setDayInitialFocused(false);
+    } else if (type === DateUtil.inputType.MONTH) {
+      setMonthInitialFocused(false);
+    } else {
+      setYearInitialFocused(false);
+    }
```

Comment:
To avoid checking the `type` twice here, would this be easier to read?
```suggestion
    let day = date.day;
    let month = date.month;
    let year = date.year;

    if (type === DateUtil.inputType.DAY) {
      day = inputValue;
      setDayInitialFocused(false);
    } else if (type === DateUtil.inputType.MONTH) {
      month = inputValue;
      setMonthInitialFocused(false);
    } else {
      year = inputValue;
      setYearInitialFocused(false);
    }
```

Reply:
Updated here : https://github.com/cerner/terra-framework/pull/1208/commits/623265a07fb2323e62a269e3e72b183ee8c8b2da